querydsl-jpa/src/main/java/com/querydsl/jpa/Hibernate5Templates.java

```diff
@@ -0,0 +1,30 @@
+package com.querydsl.jpa;
+
+/**
+ * Hibernate5Templates extends {@link JPQLTemplates} with Hibernate specific extensions
+ *
+ * @author Jan-Willem Gmelig Meyling
+ *
+ */
+public class Hibernate5Templates extends HQLTemplates {
+
+    public static final Hibernate5Templates DEFAULT = new Hibernate5Templates();
+
+    public Hibernate5Templates() {
+    }
+
+    public Hibernate5Templates(char escape) {
+        super(escape);
+    }
+
+    @Override
+    public boolean isWithForOn() {
+        return false;
```

Comment:
Hibernates WITH clause was HQL's precesessor of JPA's ON clause. Hibernate 5.x implements JPA 2.1 which introcued the ON clause for JPA. Nowadays, the WITH clause has limited functionality over the ON clause. For example entity joins require the ON clause instead. So currently, Hibernate users would actually be better of using the standard `JPATemplates` than the `JPQLTemplates`. 

This aims to set that straight.